e2e/test/iothub/SasCredentialAuthenticationTests.cs

```diff
@@ -0,0 +1,225 @@
+
+            var device = new Device(Guid.NewGuid().ToString());
+
+            // act
+            Device createdDevice = await registryManager.AddDeviceAsync(device).ConfigureAwait(false);
+
+            // assert
+            Assert.IsNotNull(createdDevice);
+
+            // cleanup
+            await registryManager.RemoveDeviceAsync(device.Id).ConfigureAwait(false);
+        }
+
+        [Ignore]
+        [LoggedTestMethod]
+        public async Task RegistryManager_Http_SasCredentialAuth_Renewed_Success()
+        {
+            // arrange
+            string signature = Configuration.IoTHub.GetIotHubSharedAccessSignature(TimeSpan.FromHours(-1));
+            var sasCredential = new AzureSasCredential(signature);
+            using var registryManager = RegistryManager.Create(
+                Configuration.IoTHub.GetIotHubHostName(),
+                sasCredential);
+
+            var device = new Device(Guid.NewGuid().ToString());
+
+            // act
+            try
+            {
+                await registryManager.AddDeviceAsync(device).ConfigureAwait(false);
+            }
+            catch (Exception)
+            {
+                // TODO: Test and catch the correct exception.
+            }
+            signature = Configuration.IoTHub.GetIotHubSharedAccessSignature(TimeSpan.FromHours(1));
+            sasCredential.Update(signature);
+            Device createdDevice = await registryManager.AddDeviceAsync(device).ConfigureAwait(false);
+
+            // assert
+            Assert.IsNotNull(createdDevice);
+
+            // cleanup
+            await registryManager.RemoveDeviceAsync(device.Id).ConfigureAwait(false);
+        }
+
+        [Ignore]
+        [LoggedTestMethod]
+        public async Task JobClient_Http_SasCredentialAuth_Success()
+        {
+            // arrange
+            string signature = Configuration.IoTHub.GetIotHubSharedAccessSignature(TimeSpan.FromHours(1));
+            using var jobClient = JobClient.Create(
+                Configuration.IoTHub.GetIotHubHostName(),
+                new AzureSasCredential(signature));
+
+            string jobId = "JOBSAMPLE" + Guid.NewGuid().ToString();
+            string jobDeviceId = "JobsSample_Device";
+            string query = $"DeviceId IN ['{jobDeviceId}']";
+            Twin twin = new Twin(jobDeviceId);
+
+            try
+            {
+                // act
+                JobResponse createJobResponse = await jobClient
+                    .ScheduleTwinUpdateAsync(
+                        jobId,
+                        query,
+                        twin,
+                        DateTime.UtcNow,
+                        (long)TimeSpan.FromMinutes(2).TotalSeconds)
+                    .ConfigureAwait(false);
+            }
+            catch (ThrottlingException)
+            {
+                // Concurrent jobs can be rejected, but it still means authentication was successful. Ignore the exception.
```

Comment:
I wonder about this one, actually. Do you think the service actually validates the auth of a request before throttling? I'd expect it not to since validating the auth seems expensive

Reply:
To add on to this, our test hubs only allow a few concurrent jobs at once, so I'd expect this exception frequently. I'd be wary of assuming this is a success case

Reply:
The service will not even receive the request if the auth fails. And the throttling is implemented in the service. So this would most definitely happen.